centreon/www/class/centreonAuth.class.php

```diff
@@ -244,6 +244,19 @@ private function checkLdapPassword($password, $autoImport): void
         }
 
         foreach ($authResources as $arId) {
+            // only try to log in using the ldap config that has been used to import the user if it still exists
+            // we check array key 0 because if the ldap config exist and match the contact ar id it defaults to index 0.
+            if ($this->userInfos['ar_id'] != $arId && array_key_exists(0, $authResources)) {
```

Comment:
if  **$this->userInfos['ar_id']**  and **$arId**, are the same type use !== operator, otherwise you can leave it like this

Reply:
you're totally right. they are both int(11) in the centreon.contact and centreon.auth_resource table so yeah strict condition is doable and now done